### PR TITLE
More tests

### PR DESCRIPTION
**Pull request by Licenser**
Add tests for code that isn't properly covered.

### CODE REVIEW

**Comment by sunnygleason** [comment]
A couple "interesting" things I'm observing:

* internal macros are being marked as untouched `stage2.rs`
* the `unsafe` leading line of an `unsafe` block is being marked as uncovered
* constants appear to be uncovered in at least a few cases as well...

**Comment by Licenser** [comment]
Ja, I saw some of the same. In addition declaration without initialisation isn't covered either. It seems like rust code coverage is not all there yet, perhaps we should aim for a lower number then 90% given how much code is counted as not covered.

**Comment by Licenser** [comment]
I added a ticket for a few of those: https://github.com/xd009642/tarpaulin/issues/262 if I missed any feel free to add them

**Comment by sunnygleason** [comment]
@Licenser oh nice, that's awesome!

I didn't look into it that much -- do the macros we're using need the very tight/deep scope? Was wondering if they could be top-level decls...

**Comment by Licenser** [comment]
Nah they need to exist in the function as they capture scpoe.

File: src/value/borrowed.rs [hunk 1]
```diff
@@ -284,6 +284,7 @@ impl<'de> BorrowDeserializer<'de> {
 
 #[cfg(test)]
 mod test {
+    #![allow(clippy::cognitive_complexity)]
```
Comment (sunnygleason) [inline src/value/borrowed.rs]: 😂 🍺

Comment (Licenser) [reply src/value/borrowed.rs]: Yap sadly clippy doesn't understand context -.-

**Pull request merged**